yt_dlp/extractor/japandiet.py

```diff
@@ -0,0 +1,259 @@
+
+
+class ShugiinItvLiveRoomIE(ShugiinItvBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?shugiintv\.go\.jp/(?:jp|en)/index\.php\?room_id=(?P<id>room\d+)'
+    IE_DESC = '衆議院インターネット審議中継 (中継)'
+
+    def _real_extract(self, url):
+        url, smug = unsmuggle_url(url)
+        if smug:
+            room_id, title = smug
+        else:
+            room_id = self._match_id(url)
+            webpage = self._download_webpage(
+                'https://www.shugiintv.go.jp/jp/index.php', room_id,
+                encoding='euc-jp', note='Looking up for the title')
+            title = traverse_obj(self._find_rooms(webpage), (lambda k, v: v['id'] == room_id, 'title'), get_all=False)
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            f'https://hlslive.shugiintv.go.jp/{room_id}/amlst:{room_id}/playlist.m3u8',
+            room_id, ext='mp4')
+        self._sort_formats(formats)
+
+        return {
+            'id': room_id,
+            'title': title,
+            'formats': formats,
+            'subtitles': subtitles,
+            'is_live': True,
+        }
+
+
+class ShugiinItvVodIE(ShugiinItvBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?shugiintv\.go\.jp/(?:jp|en)/index\.php\?ex=VL(?:\&[^=]+=[^&]*)*\&deli_id=(?P<id>\d+)'
+    IE_DESC = '衆議院インターネット審議中継 (ビデオライブラリ)'
+    _TESTS = [{
+        'url': 'https://www.shugiintv.go.jp/jp/index.php?ex=VL&media_type=&deli_id=53846',
+        'info_dict': {
+            'id': '53846',
+            'title': 'ウクライナ大統領国会演説（オンライン）',
+            'release_date': '20220323',
+            'chapters': 'count:4',
+        }
+    }, {
+        'url': 'https://www.shugiintv.go.jp/en/index.php?ex=VL&media_type=&deli_id=53846',
+        'only_matching': True
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(
+            f'https://www.shugiintv.go.jp/jp/index.php?ex=VL&media_type=&deli_id={video_id}', video_id,
+            encoding='euc-jp')
+
+        m3u8_url = self._search_regex(
+            r'id="vtag_src_base_vod"\s*value="(http.+?\.m3u8)"', webpage, 'm3u8 url')
+        m3u8_url = re.sub(r'^http://', 'https://', m3u8_url)
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            m3u8_url, video_id, ext='mp4')
+        self._sort_formats(formats)
+
+        title = self._html_search_regex(
+            (r'<td\s+align="left">(.+)\s*\(\d+分\)',
+             r'<TD.+?<IMG\s*src=".+?/spacer\.gif".+?height="15">(.+?)<IMG'), webpage, 'title', fatal=False)
+
+        release_date = self._parse_japanese_date(self._html_search_regex(
+            r'開会日</td>\s*<td.+?/td>\s*<TD>(.+?)</TD>',
+            webpage, 'title', fatal=False))
+
+        # NOTE: chapters are sparse, because of how the website serves the video
+        chapters = []
+        for chp in re.finditer(r'<A\s+HREF=".+?php\?.+?&deli_id=\d+&time=([\d\.]+)"\s*class="play_vod">(?!<img)(.+)</[Aa]>', webpage):
```

Comment:
```suggestion
        for chp in re.finditer(r'(?i)<a\s+href="[^"]+php\?[^"]+?&deli_id=\d+&time=([\d.]+)"\s*class="play_vod">(?!<img)(.+?)</a>', webpage):
```
Capture the whole URL and parse query later imo. Currently this breaks if order of query params change